include/EVT/io/I2C.hpp

```diff
@@ -3,6 +3,14 @@
 
 #include <stdint.h>
 
+#define I2C_RETURN_IF_ERR(func)             \
+    {                                       \
+        I2C::I2CStatus status = func;       \
+        if (status != I2C::I2CStatus::OK) { \
+            return status;                  \
+        }                                   \
+    }
+
```

Comment:
Is there any particular reason you opted for a macro here vs. a traditional function definition?

Reply:
Yeah, so basically I want some functions to return a value on error. So if I had this as a function, the function will return the status to the caller, but I want the caller itself to return the status on error. 

```
I2C::I2CStatus myFunc() {
    ...
    I2C_RETURN_IF_ERR(read(...));
    ...
}
```

As a macro, `myFunc()` will return the status if the read fails. Basically it leads to cleaner looking code so I don't have to constantly wrap the I2C calls in if statements.

Additionally, since its a small snippet, the argument could be made about function overhead blah blah blah, bunch of stuff we usually don't worry too much about. Really it's more about what I have above

Reply:
Ah, that makes perfect sense.